mypy/test/data.py

```diff
@@ -137,19 +142,29 @@ def add_dirs(self, dir: str) -> List[str]:
             os.mkdir(dir)
             return dirs
 
-    def run(self):
-        if self.name.endswith('-skip'):
-            raise SkipTestCaseException()
+    def run(self, obj: Any = None):
+        if obj is None:
+            # XXX: The unit tests are being converted over to pytest. Due to
+            # modifications requires to make BOTH run at the moment, this branch
+            # is necessary. It should be removed once all the tests relying on
+            # DataDrivenTestCase are converted to pytest.
+            if self.is_skip:
+                raise SkipTestCaseException()
+            else:
+                self.perform(self)
         else:
-            self.perform(self)
+            assert not self.is_skip
+            self.perform(obj, self)
 
     def tear_down(self) -> None:
         # First remove files.
         for is_dir, path in reversed(self.clean_up):
-            if not is_dir:
+            if not is_dir and os.path.exists(path):
```

Comment:
I see. Sounds like a good bug to fix!

Would you send this as a separate, tiny PR? That way it can have its own commit message that explains it's fixing this bug, and also it doesn't sit here in this unrelated PR (and ultimately this commit in the final history) causing the reader to wonder what purpose it serves and what mysterious interaction with pytest brings it into this change.

I expect that separate PR can be merged very quickly, too. :-)